src/runtime/virtcontainers/qemu.go

```diff
@@ -170,6 +170,13 @@ func (q *qemu) kernelParameters() string {
 	// set the maximum number of vCPUs
 	params = append(params, Param{"nr_cpus", fmt.Sprintf("%d", q.config.DefaultMaxVCPUs)})
 
+	// set the location of the online-kbs for SEV(-ES) guest attestation
+	if q.arch.guestProtection() == sevProtection &&
+		q.config.GuestPreAttestation &&
+		q.config.GuestPreAttestationSecretType == "connection" {
+		params = append(params, Param{"agent.aa_kbc_params", "online_sev_kbc::" + q.config.GuestPreAttestationProxy})
```

Comment:
Will the "online_sev_kbc" be the only supported "connection" type?
I'm wondering here whether we should explicitly have a few for specifying the kbc as a config, and then using that config here.

Reply:
Yes. We can probably replace `GuestPreAttestationSecretType` with `GuestPreAttestationKbsType` in the config.   

e.g., `guest_pre_attestation_kbs = "online"`

Reply:
@fitzthum what do you think? I know of the offline/simple kbs, are there plans for more in the pipeline?

Reply:
At the moment there is only `simple-kbs`, which can do both online and offline modes. It might make sense to have a more generic configuration option that allows us to select `simple-kbs` as one of many possible options. Since there are no other options idk how much of a priority this is. It doesn't seem like the generic KBS is going to support pre-attestation, but we could always add support here for it in the future.

At the moment the config can be a little bit confusing.

The connection type is a property of the `simple-kbs`. So it's not that `simple-kbs` is a connection type. Rather we are assuming from the start that you are using `simple-kbs` and that supports two different secret types (for online and offline). Since the secret types of `bundle` and `connection` map directly to the offline and online kbc, we could potentially change the config flag to select an online or offline kbc directly. This is what Jim is suggesting and it might be more clear. We could also use that to select between the two guids that we use to identify the secret.

Reply:
Ok, so bundle/connection is effectively the same as offline/online. I think changing it to offline/online will make the config clearer.  Also, I'd like to move the guids out of the config since they're bound to the KBS type (and I don't foresee that becoming more variable).

Reply:
> I think changing it to offline/online will make the config clearer. 

Or should it be local/remote?

Reply:
No. They are both remote. We've been using the online/offline terms in the AA from the start.

Reply:
I think there is technically something to having the guids be configurable separately, but it's a really marginal use case.

Reply:
So marginal that I think we can just forget about it and use one flag for both things.